api/src/main/java/org/apache/iceberg/transforms/Bucket.java

```diff
@@ -114,6 +125,7 @@ public String toString() {
 
   @Override
   public UnboundPredicate<Integer> project(String name, BoundPredicate<T> predicate) {
+    Function<T, Integer> apply = this.bind(predicate.term().type());
```

Comment:
nit: should we call this `function`? then we can avoid the `apple.apple` later in line 137, which looks a little weird.

Reply:
Done.